development/process.md

```diff
@@ -101,6 +102,93 @@ keep the amount of rework from the contributor to a minimum.
       website are updated.
     * Finally the release is announced.
 
+## Pull request and commit guidelines <a name="commit"></a>
+
+Contributions to Trino projects are managed as [collaboration with pull
+requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests).
+
+Pull requests are usually merged into `master` or `main` using the  [`rebase and
+merge`](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#rebase-and-merge-your-pull-request-commits)
+strategy.
+
+A typical pull request should strive to contain a single logical change, but not
+necessarily a single commit. Unrelated changes should generally be extracted
+into their own PRs.
+
+If a pull request contains a stack of more than one commit, then popping any
+number of commits from the top of the stack, should not break the PR, meaning
+that  every commit should build and pass all tests.
```

Comment:
It is good end goal - but given we do not check that on CI pretty sure not the case quite often (the tests part; we check compilation)

Reply:
yeah .. but mistakes happen .. thats fine .. here we should still talk about the goal..